CONTRIBUTING.md

```diff
@@ -74,18 +121,46 @@ var/path_type = /obj/item/weapon/baseball_bat
 var/path_type = "/obj/item/weapon/baseball_bat"
 ```
 
-* You must use tabs to indent your code, NOT SPACES.
+###Tabs not spaces
+You must use tabs to indent your code, NOT SPACES.
+
+(You may use spaces to align something, but you should tab to the block level first, then add the remaining spaces)
+
+###No Hacky code
+Hacky code, such as adding specific checks, is highly discouraged and only allowed when there is no other option.
+
+You can avoid hacky code by using object oriented methodologies, such as overriding a function (called procs in DM) or sectioning code into functions and then overriding them as required.
+
+###No duplicated code.
+Duplicated code is 99% of the time never allowed. 
+
+Copying code from one place to another maybe suitable for small short time projects but /tg/station focuses on the long term and thus discourages this.
+
+Instead you can use object orientation, or simply placing repeated code in a function, to obey this specification easily.
+
+###Develop Secure Code
+
+* Player input must always be escaped safely, we recommend you use stripped_input in all cases where you would use input. Essentially, just always treat input from players as inherently malicious and design with that use case in mind
```

Comment:
Someone else should check my logic, but shouldn't we only sanitize such input right before it gets rendered into HTML and sent to the client?  Double sanitizing is why `&#39;` shows up all over the place.